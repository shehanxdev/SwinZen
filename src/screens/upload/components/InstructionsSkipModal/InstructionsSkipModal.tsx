import React from 'react';
import { View } from 'react-native';

import { ModalLayout, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, TextVariant } from '@sz/constants';

interface InstructionsSkipModalProps {
  testID?: string;
  showModal: boolean;
  handleModalClose: () => void;
  onSkipped: () => void;
}

export function InstructionsSkipModal({
  testID = 'PlanSubscriptionCardTestID',
  showModal,
  handleModalClose,
  onSkipped,
}: InstructionsSkipModalProps) {
  return (
    <View testID={testID}>
      <ModalLayout
        showModal={showModal}
        handleModalClose={handleModalClose}
        leftButtonTitle="YES"
        rightButtonTitle="NO"
        onLeftButtonPress={onSkipped}
        onRightButtonPress={handleModalClose}>
        <View style={tw`mb-3 px-12.5`}>
          <Text variant={TextVariant.Body1SemiBold} color={Color.Neutral.White}>
            Always skip instructions?
          </Text>
        </View>
        <View style={tw`mb-11 p-[10]`}>
          <Text variant={TextVariant.Body1Regular}>
            If you want to skip the instructions with future uploads, select YES. If you want to keep seeing the
            instructions with every upload, select NO.
          </Text>
        </View>
      </ModalLayout>
    </View>
  );
}
