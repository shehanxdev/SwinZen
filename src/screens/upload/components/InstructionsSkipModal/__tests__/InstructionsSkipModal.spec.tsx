import { render } from '@testing-library/react-native';
import * as React from 'react';

import { InstructionsSkipModal } from '../InstructionsSkipModal';

describe('InstructionsSkipModal Component', () => {
  const testID = 'InstructionsSkipModalTestID';
  const mockFn = jest.fn();

  const getRenderedComponent = () =>
    render(<InstructionsSkipModal testID={testID} showModal={false} handleModalClose={mockFn} onSkipped={mockFn} />);

  it(`should render InstructionsSkipModal component correctly`, () => {
    const rendered = getRenderedComponent();
    const renderedTree = rendered.toJSON();
    expect(renderedTree).toMatchSnapshot();
  });

  it(`should find the InstructionsSkipModal component via testID`, () => {
    const { getByTestId } = getRenderedComponent();
    const foundCustomHeaderComponent = getByTestId(testID);
    expect(foundCustomHeaderComponent).toBeTruthy();
  });
});
