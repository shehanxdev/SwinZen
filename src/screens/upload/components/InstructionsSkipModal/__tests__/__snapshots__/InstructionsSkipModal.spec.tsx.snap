// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructionsSkipModal Component should render InstructionsSkipModal component correctly 1`] = `
<View
  testID="InstructionsSkipModalTestID"
>
  <Modal
    animationType="none"
    deviceHeight={null}
    deviceWidth={null}
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    onRequestClose={[MockFunction]}
    panResponderThreshold={4}
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={false}
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  />
</View>
`;
